runtime/binding-mqtt/src/main/zilla/protocol.idl

```diff
@@ -94,26 +94,26 @@
             case 0x2A: uint8 sharedSubscriptionAvailable;
         }
 
-        struct MqttConnectV3 extends MqttPacketHeader
+        struct MqttConnectHeader extends MqttPacketHeader
         {
             string16 protocolName;
             uint8 protocolVersion;
             uint8 flags;
             uint16 keepAlive;
+        }
+
+        struct MqttConnectV4
+        {
             string16 clientId;
         }
 
-        struct MqttConnectV5 extends MqttPacketHeader
+        struct MqttConnectV5
```

Comment:
Perhaps `MqttConnectV4` and `MqttConnectV5` can extend `MqttConnectHeader` so it is not necessary to think about the separate parts when sending, such as from mqtt client.

Reply:
Yes, that is understood.

However you can have both the separate `MqttConnect` struct and the `MqttConnectV5 extends MqttConnect` struct (also `MqttConnectV4 extends MqttConnect` struct) to eliminate the unnatural creating of the `MqttConnectV5` flyweight in 2 parts when sending from mqtt client.